backend/app/authentication.py

```diff
@@ -56,17 +56,15 @@ async def _get_user(
         """
         if not from_token and not from_api_key:
             raise ValueError("Either from_token or from_api_key must be set to True")
-
-        if not required and not token and not api_key:
+        elif not required and not token and not api_key:
             return None
-
-        if from_token and token:
+        elif from_token and token:
             return await _verify_token(token)
-        if from_api_key and api_key:
+        elif from_api_key and api_key:
             # TODO: Fetch userinfo based on user email from Keycloak
             # needs special role/rights in Keycloak for the client
             # In this way, this will return the same user info.
-            user_obj = await RailUser.find_one({"api_key": api_key})
+            user_obj = await RailUser.find_one(RailUser.api_key == api_key)
```

Comment:
Tests are failing when mocking with this syntax. Lets change it to the one used before `find_one({"api_key": api_key})`